joy_teleop/joy_teleop/joy_teleop.py

```diff
@@ -59,52 +330,47 @@ def __init__(self):
         super().__init__('joy_teleop', allow_undeclared_parameters=True,
                          automatically_declare_parameters_from_overrides=True)
 
-        self.config = {}
-        self.pubs = {}
-        self.al_clients = {}
-        self.srv_clients = {}
-        self.message_types = {}
-        self.command_list = {}
-        self.offline_actions = []
-        self.offline_services = []
-
-        self.old_buttons = []
+        self.commands = []
 
-        self.retrieve_config()
+        names = []
 
-        for i, config in self.config.items():
-            if i in self.command_list:
-                self.get_logger().error('command {} was duplicated'.format(i))
-                continue
+        for name, config in self.retrieve_config().items():
+            if name in names:
+                raise JoyTeleopException("command '{}' was duplicated".format(name))
 
             try:
                 interface_group = config['type']
 
-                self.add_command(i, config)
-
                 if interface_group == 'topic':
-                    self.register_topic(i, config)
-                elif interface_group == 'action':
-                    self.register_action(i, config)
+                    self.commands.append(JoyTeleopTopicCommand(name, config, self))
                 elif interface_group == 'service':
-                    self.register_service(i, config)
+                    self.commands.append(JoyTeleopServiceCommand(name, config, self))
+                elif interface_group == 'action':
+                    self.commands.append(JoyTeleopActionCommand(name, config, self))
                 else:
-                    self.get_logger().error("unknown type '{type}'"
-                                            "for command '{i}'".format_map(locals()))
+                    raise JoyTeleopException("unknown type '{interface_group}' "
+                                             "for command '{name}'".format_map(locals()))
             except TypeError:
-                self.get_logger().warn(f"parameter {i} is not a dict")
+                # This can happen on parameters we don't control, like
+                # 'use_sim_time'.  Just ignore it with a warning for the user.
+                self.get_logger().info('parameter {} is not a dict'.format(name))
```

Comment:
@clalancette I'd say the comment about not making any noise when something expected comes you way still stands, but it's not a strong stand.

Reply:
I was kind of leaving this in since it was already there, and it is a warning to the user if they entered in something that they didn't mean.  But I think we have enough other error checking now that it isn't needed as much.  So I reduced this even further to a `debug`, so you won't see it by default.